test/test-v8debugapi.js

```diff
@@ -727,65 +735,139 @@ describe('v8debugapi', function() {
-      });
     });
 
-    it('should limit object length', function(done) {
-      var bp = {
-        id: 'fake-id-124',
-        location: { path: 'test-v8debugapi.js', line: 5 },
-        expressions: ['B']
-      };
-      var oldMax = config.capture.maxProperties;
-      config.capture.maxProperties = 1;
-      api.set(bp, function(err) {
-        assert.ifError(err);
-        api.wait(bp, function(err) {
+    it('should display an error for an evaluated array beyond maxDataSize',
+      function(done) {
+        var bp = {
+          id: 'fake-id-124',
+          location: { path: 'test-v8debugapi.js', line: 5 },
+          expressions: ['A']
+        };
+        var oldMaxProps = config.capture.maxProperties;
+        var oldMaxData = config.capture.maxDataSize;
+        config.capture.maxProperties = 5;
+        config.capture.maxDataSize = 1;
+        api.set(bp, function(err) {
+          assert.ifError(err);
+          api.wait(bp, function(err) {
+            assert.ifError(err);
+            var foo = bp.evaluatedExpressions[0];
+            var fooVal = bp.variableTable[foo.varTableIndex];
+            assert(fooVal.status.description.format.indexOf(
+              'Max data size reached') !== -1);
+            assert(fooVal.status.isError);
+
+            api.clear(bp);
+            config.capture.maxDataSize = oldMaxData;
+            config.capture.maxProperties = oldMaxProps;
+            done();
+          });
+          process.nextTick(function() {foo(2);});
+        });
+    });
+
+    it('should not limit the length of an evaluated object based on maxProperties',
+      function(done) {
+        var bp = {
+          id: 'fake-id-124',
+          location: { path: 'test-v8debugapi.js', line: 5 },
+          expressions: ['B']
+        };
+        var oldMaxProps = config.capture.maxProperties;
+        var oldMaxData = config.capture.maxDataSize;
+        config.capture.maxProperties = 1;
+        config.capture.maxDataSize = 20000;
+        api.set(bp, function(err) {
           assert.ifError(err);
-          var foo = bp.evaluatedExpressions[0];
-          var fooVal = bp.variableTable[foo.varTableIndex];
-          assert.equal(fooVal.members.length, 1);
-          assert(foo.status.description.format.indexOf(
-            'Only first') !== -1);
-          assert(!foo.status.isError);
+          api.wait(bp, function(err) {
+            assert.ifError(err);
+            var foo = bp.evaluatedExpressions[0];
+            var fooVal = bp.variableTable[foo.varTableIndex];
+            assert.equal(fooVal.members.length, 3);
+            assert.strictEqual(foo.status, undefined);
 
-          api.clear(bp);
-          config.capture.maxProperties = oldMax;
-          done();
+            api.clear(bp);
+            config.capture.maxDataSize = oldMaxData;
+            config.capture.maxProperties = oldMaxProps;
+            done();
+          });
+          process.nextTick(function() {foo(2);});
```

Comment:
The existing tests used process.nextTick rather than setImmediate. Before making a change, I want to make sure the use of process.nextTick wasn't a necessity in this particular case.